### ISSUE TITLE
[Question] error 

### ISSUE THREAD

**Issue by jongyook2** [msu-coinlab/pymoo#24]
[-2.21617357e-02 -1.50119180e-01 -9.09995681e-02  1.25947932e-01
  -6.15241812e-02 -2.23912419e-01  5.85929008e-01 -1.31129174e-01
  -4.12985813e-01 -5.06295108e-01  2.03546639e-01  5.36321300e-02
  -2.33383690e-01  1.14347371e-01  4.88084609e-01 -3.17626463e-02
   4.24874150e-02 -2.58434832e-01  1.88217357e-02  1.38999180e-01
   7.75795681e-02 -1.47667932e-01  4.32841812e-02  2.16312419e-01
  -5.91369008e-01  1.17529174e-01  4.03005813e-01  4.89135108e-01
  -2.19946639e-01 -5.98121300e-02  2.18683690e-01 -1.40927371e-01
  -5.21544609e-01  2.31226463e-02 -5.27274150e-02  2.39014832e-01]]
10    | 230     | 3.3645104619 / 3.7806289242 | -            | -           
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-18-e017f7e4988c> in <module>
     13                pf=None,
     14                save_history=True,
---> 15                verbose=True)
     16 # res = minimize(problem,algorithm,termination)

C:\Users\Farmboy\Anaconda3\envs\flavor\lib\site-packages\pymoo\optimize.py in minimize(problem, algorithm, termination, **kwargs)
     56 
     57     # actually execute the algorithm
---> 58     res = algorithm.solve()
     59 
     60     # store the copied algorithm in the result object

C:\Users\Farmboy\Anaconda3\envs\flavor\lib\site-packages\pymoo\model\algorithm.py in solve(self)
    169             X, F, CV, G = opt.get("X", "F", "CV", "G")
    170         else:
--> 171             X, F, CV, G = opt.X, opt.F, opt.CV, opt.G
    172 
    173         if opt is not None:

AttributeError: 'NoneType' object has no attribute 'X'

**This is my code** 
class MyProblem(Problem):

    def __init__(self):
        super().__init__(n_var=2,
                         n_obj=1,
                         n_constr=36,
                         xl=anp.array([0,0]),
                         xu=anp.array([100,100]))
    
       
    def _evaluate(self, x, out, *args, **kwargs):
        f1 = x[:,0] + x[:,1]      
        out["F"] =anp.column_stack([f1])
        out["G"] =anp.column_stack([out_cal(temp_c1,temp_c2,temp_c3,x),out_cal_neg(temp_c1_neg,temp_c2_neg,temp_c3,x)])
        print(out["G"])
```
from pymoo.algorithms.nsga2 import NSGA2
from pymoo.factory import get_algorithm, get_sampling, get_crossover, get_mutation
from pymoo.model.repair import Repair    
algorithm = NSGA2(
    pop_size=50,
    n_offsprings=20,
    sampling=get_sampling("real_random"),
    crossover=get_crossover("real_sbx", prob=0.9, eta=15),
    mutation=get_mutation("real_pm", eta=20),
    eliminate_duplicates=True
)
from pymoo.factory import get_termination

termination = get_termination("n_gen", 10)
import matplotlib.pyplot as plt

from pymoo.optimize import minimize
from pymoo.util import plotting

res = minimize(problem,
               algorithm,
               termination,
               seed=1,
               pf=None,
               save_history=True,
               verbose=True)
```
obj: 1 constr:36 var:2 
I can check that out["G"] has 36 elements, but it shows the same error like last issue(pf=True).
Thank you in advance.

**Comment by julesy89** [comment]
I do not know what you out_cal and out_cal_neg do, but I strongly believe that the algorithm was no able to find any feasible solution. I realized that in this case an exception is thrown which should not be the case.

The algorithm should either return None or return the least feasible solution. I fixed the bug in my last commit. There is now an option

`NSGA2(return_least_infeasible=False)`

which gives you an additional choice of the behavior if no feasible solution was found.
Pull the latest master branch and install to check if it works for you. I am planning to release a new pymoo version next or next next week.

3d25b1692ed6badbfad1c421383bfa1df770439d
